test/test-style.js

```diff
@@ -163,6 +163,14 @@ function testStyle(filename) {
     console.error('\x1b[33m  Style – Found \\" but expected \' for <a href>.\x1b[0m');
   }
 
+  let relativeUrlMatch = actual.match(String.raw`<a href='(?!www\.|(?:http)s?:\/\/|[A-Za-z]:\\|\/\/)(.*)'>`);
```

Comment:
I *think* this regex would work fine for our purposes. I wonder why you escaped the slashes? (`\/\/ `instead of `//` [two times]). Fwiw, here's some interesting geekery on URL regexes https://mathiasbynens.be/demo/url-regex

Alternatively, can we use the URL API that is available to node as well, I think? https://nodejs.org/api/url.html

Reply:
The regex, I copied directly from #2029, and it seemed to work as intended so I didn't bother to modify those parts.  Looking at the URL API, I'm not sure if it'll suit the purposes of detecting relative URLs, but I'm happy to look into it further and switch if desired!

Reply:
@vinyldarkscratch I dealt with this for another project recently. Something to point you in the right direction:

```node
const url = require('url');

const abs_url = url.parse('https://developer.mozilla.org');
console.log(abs_url.hostname === null);

const rel_url = url.parse('/some/relative/path/');
console.log(rel_url.hostname === null);
```

Reply:
Well hey, there we are!  It looks far cleaner than the massive blob of regex -- will get to implementing it ASAP.  :wink:

Reply:
And done, thanks for sharing this with me, @ddbeck!